src/webui/webapplication.cpp

```diff
@@ -651,7 +672,7 @@ bool WebApplication::isCrossSiteRequest(const Http::Request &request) const
         const bool isInvalid = !isSameOrigin(urlFromHostHeader(targetOrigin), originValue);
         if (isInvalid)
             LogMsg(tr("WebUI: Origin header & Target origin mismatch! Source IP: '%1'. Origin header: '%2'. Target origin: '%3'")
-                   .arg(m_env.clientAddress.toString(), originValue, targetOrigin)
+                   .arg(m_clientAddress.toString(), originValue, targetOrigin)
```

Comment:
Maybe I'm wrong but I don't think this is correct. The log message in this context is supposed to show the client address of the connection and not the one in `X_FORWARDED_FOR`. If done otherwise, it would be very confusing when reading the log message.
Same for the changes in `WebApplication::validateHostHeader()`.

Reply:
Hmm I think you're right.

Reply:
But for `WebApplication::validateHostHeader()` I think it depends on where does the Host Header came from, which is the client in `X_FORWARDED_FOR` in most cases.
Maybe log both address when reverse proxy is enabled ?

Reply:
>I think it depends on where does the Host Header came from, which is the client in X_FORWARDED_FOR in most cases.

AFAIK in our scenario here, the Host header is generated/specified by the reversed proxy since the client is connecting to it (and not to qbt server directly). If that is the case, then there is no point of adding  `X_FORWARDED_FOR` in log for this kind of error.